plugin/pkg/admission/nodeenv/admission.go

```diff
@@ -0,0 +1,187 @@
+package nodeenv
+
+import (
+	"fmt"
+	"io"
+	"reflect"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/yaml"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+var NamespaceNodeSelectors = []string{"kubernetes.io/node-selector"}
```

Comment:
> Public variables are bad style.

This needs to be available as a plug point downstream for us to support our existing annotation.  Kube lacks the config concept like https://github.com/openshift/origin/blob/master/pkg/cmd/server/api/types.go#L1175, so actually using the controller config file is comparatively difficult.